src/todo/TodoPage.js

```diff
@@ -0,0 +1,92 @@
+import { Component } from 'react';
+import './TodoPage.css';
+import {
+  addTodo,
+  getTodos,
+  updateTodoCompleted,
+  deleteTodo,
+} from '../utils/todo-api.js';
+
+export default class TodoPage extends Component {
+  state = {
+    todos: [],
+    task: '',
+  };
+
+  componentDidMount = async () => {
+    const updatedTodos = await getTodos();
+    console.log(updatedTodos);
+    this.setState({ todos: updatedTodos });
+  };
+
+  handleNewTodo = ({ target }) => {
+    this.setState({ task: target.value });
+  };
+
+  handleAddTodo = async (e) => {
+    e.preventDefault();
+    const { task, todos } = this.state;
+    try {
+      if (task) {
+        const todo = await addTodo({
+          task: task,
+          completed: false,
+          shared: false,
+        });
+        const updatedTodos = [...todos, todo];
+        this.setState({ task: '', todos: updatedTodos });
+      } else {
+        console.log('error: invalid todo');
+      }
+    } catch (err) {
+      console.log(err.message);
+    }
+  };
+
+  toggleComplete = async (updatedTodo) => {
+    updatedTodo.completed = !updatedTodo.completed;
+    const updatedDatabaseTodo = await updateTodoCompleted(updatedTodo);
+    const updatedTodos = this.state.todos.map((todo) => {
```

Comment:
you could destructor this.state.todos to stay consistent with the rest of the code.